spmd/compiler/fusion.py

```diff
@@ -178,6 +177,8 @@ def _scan_graph_for_fusion_elements(
     fe_size = len(fe_sequence) - 1
     index = 0
     curr_node_list = []
+    # depth to reach comm_node, adjust based on clone presence
+    offset_to_comm_node = 3
```

Comment:
A general question: is this always the fixed number? Seems to be brittle. Any way to figure it out dynamically?

Reply:
It's normally fixed, but agree it's better to just auto-adapt and not have a brittle assumption.
I've added a run once autodepth finder that seeks and stores the offset, so that now code will automatically adjust for any future changes in fe_sequence. 
Thanks for the feedback!